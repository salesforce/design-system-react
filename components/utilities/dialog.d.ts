declare module '@salesforce/design-system-react/components/utilities/dialog' {
	import React from 'react';
	type Props = {
		/**
		 * Alignment of the dialog with respect to the target (assuming left-to-right language direction). For example,
		 * a value of 'left bottom' indicates that the dialog will be rendered below and left-aligned with the target.
		 * Note that setting the direction prop to "rtl" will flip the resulting dialog alignment.
		 */
		align?:
			| 'top'
			| 'top left'
			| 'top right'
			| 'right'
			| 'right top'
			| 'right bottom'
			| 'bottom'
			| 'bottom left'
			| 'bottom right'
			| 'left'
			| 'left top'
			| 'left bottom';
		/**
		 * CSS classes to be added to the absolutely positioned element.
		 */
		className?: any[] | Record<string, any> | string;
		/**
		 * CSS classes to be added to the wrapping `div` of the contents of the dialog.
		 */
		contentsClassName?: any[] | Record<string, any> | string;
		/**
		 * Contents of dialog
		 */
		children: React.ReactNode /*.isRequired*/;
		/**
		 * Closes dialog when tab key is pressed
		 */
		closeOnTabKey?: boolean;
		/**
		 * Props passed along to wrapping div. This allows one less wrapping `div` to be in the markup. dialog children are expected to be wrapper in a single `div`.
		 */
		containerProps?: Record<string, any>;
		/**
		 * Establishes directional context for component. Defaults to left-to-right.
		 */
		direction?: DIRECTIONS.LTR | DIRECTIONS.RTL;
		/**
		 * Will show the nubbin pointing from the dialog to the reference element. Positioning and offsets will be handled.
		 */
		hasNubbin?: boolean;
		/**
		 * By default, dialogs will flip their alignment (such as bottom to top) if they extend beyond a boundary element such as a scrolling parent or a window/viewpoint. `hasStaticAlignment` disables this behavior and allows this component to extend beyond boundary elements.
		 */
		hasStaticAlignment?: boolean;
		/**
		 * Sets the dialog width to the width of either 'target' (Menus attached to `input` typically follow this UX pattern), 'menu' or 'none.
		 */
		inheritWidthOf?: 'target' | 'menu' | 'none';
		/**
		 * DEPRECATED - do not add checkProp deprecation message at this level. It is handled at higher level components.
		 * TODO - to be removed.
		 * Offset adds pixels to the absolutely positioned dropdown menu in the format: ([vertical]px [horizontal]px). SHOULD BE OBJECT -----------
		 */
		offset?: string;
		/**
		 * Called when dialog closes and unmounts.
		 */
		onClose?: (v: any) => any;
		/**
		 * Called when a key pressed.
		 */
		onKeyDown?: (v: any) => any;
		/**
		 * Called when mouse hovers over the trigger button. This is only called if `this.props.openOn` is set to `hover`.
		 */
		onMouseEnter?: (v: any) => any;
		/**
		 * Called when mouse hover leaves the trigger button. This is only called if `this.props.openOn` is set to `hover`.
		 */
		onMouseLeave?: (v: any) => any;
		/**
		 * Called when dialog opens (that is mounts). The parameters are `undefined, { portal: this.portal }`.
		 */
		onOpen?: (v: any) => any;
		/**
		 * React component to be aligned with. Function should return a DOM `ref` from the parent component.
		 */
		onRequestTargetElement: (v: any) => any /*.isRequired*/;
		/**
		 * Triggered when an item in the menu is clicked.
		 */
		outsideClickIgnoreClass?: string;
		/**
		 * If a dialog is `positione="overflowBoundaryElement"`, it will be rendered in a portal or separate render tree. This `portalMount` callback will be triggered instead of the the default `ReactDOM.unstable_renderSubtreeIntoContainer` and the function will mount the portal itself. Consider the following code that bypasses the internal mount and uses an Enzyme wrapper to mount the React root tree to the DOM.
		 *
		 * ```
		 * <Popover
		 *   isOpen
		 *   portalMount={({ instance, reactElement, domContainerNode }) => {
		 *     portalWrapper = Enzyme.mount(reactElement, { attachTo: domContainerNode });
		 *   }}
		 *   onOpen={() => {
		 *     expect(portalWrapper.find(`#my-heading`)).to.exist;
		 *     done();
		 *   }}
		 *   />
		 * ```
		 */
		portalMount?: (v: any) => any;
		/**
		 * Please select one of the following:
		 * * `absolute` - (default) The dialog will use `position: absolute` and style attributes to position itself. This allows inverted placement or flipping of the dialog.
		 * * `overflowBoundaryElement` - The dialog will overflow scrolling parents. Use on elements that are aligned to the left or right of their target and don't care about the target being within a scrolling parent. Typically this is a popover or tooltip. Dropdown menus can usually open up and down if no room exists. In order to achieve this a portal element will be created and attached to `body`. This element will render into that detached render tree.
		 * * `relative` - No styling or portals will be used. Menus will be positioned relative to their triggers. This is a great choice for HTML snapshot testing.
		 */
		position?:
			| 'absolute'
			| 'overflowBoundaryElement'
			| 'relative' /*.isRequired*/;
		/**
		 * An object of CSS styles that are applied to the immediate parent `div` of the contents. Use this instead of margin props.
		 */
		style?: Record<string, any>;

		/**
		 * Sets which focus UX pattern to follow. For instance, popovers trap focus and must be exited to regain focus. Dropdowns and Tooltips never have focus.
		 */
		variant?: 'dropdown' | 'popover' | 'tooltip';
	};
	/*
	 * A Dialog is content that is separate from the typical flow of a page. It typically overlays other elements in the document flow. This is achieved with elevation (`z-index`) and one of the following: relative position, absolute position, or a new top-level React render tree (portal). A boundary element is a scrolling ancestor element or the edge of the browser (window/viewport). This element typically has an overflow (overflow-y/overflow-x) style that is scroll, hidden, or auto. Inverted placement is the flipping of the overlay element from top to bottom or left to right in order stay within a boundary element.
	 *
	 * * Dropdown menu (Combobox, DatePicker, et al.) placement is typically bottom-aligned and should be allowed to invert its placement when inside a boundary element this often happens within a modal. Dropdowns should not overflow boundary elements , since most boundary elements scroll vertically and have space for the menu.
	 *
	 * * If they are hidden, left and right placed overlay elements (such as Popover and Tooltip) should be placed within a portal element attached to the DOM <body>, but styled to align to its target/trigger. Since scrolling typically occurs on the vertical axis, this allows them to overflow boundary elements and still allow scrolling of content, yet still invert placement for the browser viewport. Portal elements are only necessary if an original ancestor boundary element exists. **No portals are created by default.**
	 *
	 * * Nubbins/arrows should be repositioned for any new placement/alignment.
	 *
	 * Allow Overflowing of Boundary Element: Allow applications to create a portal element attached to the DOM <body> to be outside of boundary elements if manual testing shows confusing alignment/poor usability/readability. This should be exception and not the default.
	 *
	 * Allow Inverted placement: Allow applications to prevent inverted placement if manual testing shows confusing alignment/poor usability/readability.
	 *
	 * ### How this new Dialog component works
	 * * There is no longer an inline render within components. All overlays should go through `Dialog`. `position: relative` just passes on the markup with some additional event listeners. No positional library is used.
	 * * The default `position: absolute` will run through three renders. It will first render the overlay at `0px/0px` offset of its parent. It will then store the target DOM node, once the DOM node is set, a state change will occur and tell a PopperJS instance to be created on update, once it is created, a third render is done to update the styles created by PopperJS.
	 * * `position: overflowBoundaryElement` will do the same three renders as `absolute` except that the initial render will create a disconnected render tree (portal) on the `body`. Then, the position will change once the target is stored. The portal itself will be rendered multiple times. The first will result in `onOpen` executing. Each update will result in a re-render of the disconnected render-tree.
	 *
	 * This component is private.
	 */
	function Component(props: Props): JSX.Element;
	export default Component;
}
