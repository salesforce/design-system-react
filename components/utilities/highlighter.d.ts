declare module '@salesforce/design-system-react/components/utilities/highlighter' {
	import React from 'react';
	type Props = {
		/**
		 * The full string to display.
		 */
		children?: string | number | boolean | React.ReactNode;
		className?: string;
		/**
		 * The string of text (or Regular Expression) to highlight.
		 */
		search?: PropTypes.any;
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
