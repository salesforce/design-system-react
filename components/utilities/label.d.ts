declare module '@salesforce/design-system-react/components/utilities/label' {
	import React from 'react';
	type Props = {
		/*
		 * Assistive Text to use instead of a visible label
		 */
		assistiveText?: Record<string, any>;
		/**
		 * Class names to be added to the label
		 */
		className?: any[] | Record<string, any> | string;
		/*
		 * Id of the input associated with this label
		 */
		htmlFor?: string;
		/*
		 * Input Label
		 */
		label?: string;
		/*
		 * Applies label styling for a required form element
		 */
		required?: boolean;
		/**
		 * Changes markup of label.
		 */
		variant?: 'base' | 'static';
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
