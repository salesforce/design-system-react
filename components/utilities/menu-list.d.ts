declare module '@salesforce/design-system-react/components/utilities/menu-list' {
	import React from 'react';
	type Props = {
		/**
		 * Determines whether or not to show a checkmark for selected items.
		 */
		checkmark?: boolean;
		/**
		 * CSS classes to be added to `<ul />`.
		 */
		className?: string;
		/**
		 * Used internally to determine the id that will be used for list items.
		 */
		getListItemId?: (v: any) => any;
		/**
		 * Used internally to pass references to the individual menu items back up for focusing / scrolling.
		 */
		itemRefs?: (v: any) => any;
		/**
		 * If provided, this function will be used to render the contents of each menu item.
		 */
		itemRenderer?: (v: any) => any;
		/**
		 * Sets the height of the list based on the numeber of items.
		 */
		length?: null | '5' | '7' | '10' | 5 | 7 | 10;
		/**
		 * Triggered when a list item is selected (via mouse or keyboard).
		 */
		onSelect?: (v: any) => any;
		/**
		 * An array of items to render in the list.
		 */
		options?: any[];
		/**
		 * The index of the currently selected item in the list.
		 */
		selectedIndex?: number;
		/**
		 * Accepts a `Tooltip` component to be used as the template for menu item tooltips that appear via the `tooltipContent` options object attribute
		 */
		tooltipMenuItem?: React.ReactNode;
		/**
		 * The id of the element which triggered this list (in a menu context).
		 */
		triggerId?: string;
	};
	/**
	 * Component description.
	 */
	function Component(props: Props): JSX.Element;
	export default Component;
}
