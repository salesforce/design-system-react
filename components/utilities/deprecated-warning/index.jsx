/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';

// ## Constants
import { DEPRECATED_WARNING } from '../../../utilities/constants';

import Alert from '../../alert';
import AlertContainer from '../../alert/container';

/**
 * A utility component that is used to highlight a deprecated component.
 */
const DeprecatedWarning = () => (
	<div className="slds-p-top_x-large">
		<AlertContainer>
			<Alert
				labels={{
					heading: (
						<span>
							<strong>Deprecated component.</strong> No fit and finish bug fixes
							will be accepted for this component.
						</span>
					),
				}}
				variant="error"
			/>
		</AlertContainer>
	</div>
);

// ### Display Name
DeprecatedWarning.displayName = DEPRECATED_WARNING;

export default DeprecatedWarning;
