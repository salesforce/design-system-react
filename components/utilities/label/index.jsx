/* eslint-disable react/jsx-curly-brace-presence */
/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const propTypes = {
	/*
	 * Assistive Text to use instead of a visible label
	 */
	assistiveText: PropTypes.object,
	/**
	 * Class names to be added to the label
	 */
	className: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.object,
		PropTypes.string,
	]),
	/*
	 * Id of the input associated with this label
	 */
	htmlFor: PropTypes.string,
	/*
	 * Input Label
	 */
	label: PropTypes.string,
	/*
	 * Applies label styling for a required form element
	 */
	required: PropTypes.bool,
	/**
	 * Changes markup of label.
	 */
	variant: PropTypes.oneOf(['base', 'static']),
};

/*
 * Form label. This returns null if there is no label text (hidden or shown)
 */
const Label = ({
	variant = 'base',
	label,
	assistiveText,
	className,
	htmlFor,
	required,
}) => {
	const labelText = label || (assistiveText && assistiveText.label); // One of these is required to pass accessibility tests

	const subRenders = {
		base: (
			<label
				className={classNames(
					'slds-form-element__label',
					{
						'slds-assistive-text': assistiveText && !label,
					},
					className
				)}
				htmlFor={htmlFor}
			>
				{required && (
					<abbr className="slds-required" title="required">
						{'*'}
					</abbr>
				)}
				{labelText}
			</label>
		),
		static: (
			<span className={classNames('slds-form-element__label', className)}>
				{labelText}
			</span>
		),
	};

	return labelText ? subRenders[variant] : null;
};

Label.displayName = 'Label';
Label.propTypes = propTypes;

export default Label;
