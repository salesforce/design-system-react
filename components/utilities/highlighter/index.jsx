/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';
import PropTypes from 'prop-types';

// ### ReactHighlighter
import ReactHighlighter from 'react-highlighter';

// ## Constants
import { HIGHLIGHTER } from '../../../utilities/constants';

/**
 * A utility component that highlights occurrences of a particular pattern in its contents.
 */
const Highlighter = (props) => {
	if (props.search) {
		let children;
		if (typeof props.children === 'string') {
			children = (
				<ReactHighlighter
					className={props.className}
					matchClass={null}
					matchElement="mark"
					search={props.search}
					title={props.children}
				>
					{props.children}
				</ReactHighlighter>
			);
		} else {
			const findString = (nodeArr) =>
				nodeArr.map((element) => {
					let newElement;
					if (typeof element === 'string') {
						newElement = (
							<ReactHighlighter
								key={element}
								className={props.className}
								matchClass={null}
								matchElement="mark"
								search={props.search}
								title={element}
							>
								{element}
							</ReactHighlighter>
						);
					} else {
						newElement = element;
					}
					return newElement;
				});

			if (props.children.props) {
				const node = props.children.props.children;
				children = node instanceof Array ? findString(node) : node;
			}
		}

		return <span>{children}</span>;
	}

	if (typeof props.children === 'string') {
		return (
			<span className={props.className} title={props.children}>
				{props.children}
			</span>
		);
	}

	return <span className={props.className}>{props.children}</span>;
};

// ### Display Name
Highlighter.displayName = HIGHLIGHTER;

// ### Prop Types
Highlighter.propTypes = {
	/**
	 * The full string to display.
	 */
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.bool,
		PropTypes.node,
	]),
	className: PropTypes.string,
	/**
	 * The string of text (or Regular Expression) to highlight.
	 */
	search: PropTypes.any,
};

export default Highlighter;
