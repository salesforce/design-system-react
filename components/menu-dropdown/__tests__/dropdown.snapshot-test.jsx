import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	testDOMandHTML,
	testImageSnapshot
} from '../../../tests/snapshot-helpers';

import Checkmark from '../__examples__/checkmark';
import CustomTrigger from '../__examples__/custom-trigger';
import Default from '../__examples__/default';
import DefaultIconLabel from '../__examples__/default-icon-label';
import SubHeading from '../__examples__/sub-heading';

import { MENU_DROPDOWN } from '../../../utilities/constants';

describe(MENU_DROPDOWN, () => {
	test('Checkmark DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(Checkmark);
	});

	test('CustomTrigger DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(CustomTrigger);
	});

	test('Default DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(Default);
	});

	test('DefaultIconLabel DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(DefaultIconLabel);
	});

	test('SubHeading DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(SubHeading);
	});

	test('Base Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Base');
	});

	test('Base with icon Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Base with icon');
	});

	test('Render inline Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Render inline');
	});

	test('Render inline w/ Nubbins Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Render inline w/ Nubbins');
	});

	test('Custom Trigger Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Custom Trigger');
	});

	test('Custom Content Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Custom Content');
	});

	test('Hover Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Hover');
	});

	test('Two Hovers Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Two Hovers');
	});

	test('Hover with Checkmark Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Hover with Checkmark');
	});

	test('Controled w/ isOpen Image Snapshot looks the same', async () => {
		await testImageSnapshot(MENU_DROPDOWN, 'Controled w/ isOpen');
	});
});
