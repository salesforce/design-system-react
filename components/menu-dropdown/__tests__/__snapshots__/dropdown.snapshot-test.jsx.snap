// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SLDSMenuDropdown Checkmark DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center"
>
  <div
    className="slds-col--padded"
  >
    <span>
      Checkmark 
    </span>
    <div
      className="slds-dropdown-trigger slds-dropdown-trigger--click"
      id="H1iMi7aVf"
      onBlur={undefined}
      onClick={[Function]}
      onFocus={null}
      onKeyDown={[Function]}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
    >
      <button
        aria-controls={undefined}
        aria-describedby={undefined}
        aria-expanded={false}
        aria-haspopup={true}
        className="slds-button slds-button--icon-border-filled ignore-click-H1iMi7aVf"
        disabled={false}
        id={undefined}
        onBlur={undefined}
        onClick={[Function]}
        onFocus={undefined}
        onKeyDown={undefined}
        onKeyPress={undefined}
        onKeyUp={undefined}
        onMouseDown={undefined}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        tabIndex="0"
        title={undefined}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <span>
      Checkmark with right icon 
    </span>
    <div
      className="slds-dropdown-trigger slds-dropdown-trigger--click"
      id="HklszjXaVM"
      onBlur={undefined}
      onClick={[Function]}
      onFocus={null}
      onKeyDown={[Function]}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
    >
      <button
        aria-controls={undefined}
        aria-describedby={undefined}
        aria-expanded={false}
        aria-haspopup={true}
        className="slds-button slds-button--icon-more slds-button--icon-large ignore-click-HklszjXaVM"
        disabled={false}
        id={undefined}
        onBlur={undefined}
        onClick={[Function]}
        onFocus={undefined}
        onKeyDown={undefined}
        onKeyPress={undefined}
        onKeyUp={undefined}
        onMouseDown={undefined}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        tabIndex="0"
        title={undefined}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
          />
        </svg>
        <svg
          aria-hidden="true"
          className="slds-button__icon slds-button__icon--x-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Checkmark with right icon
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`SLDSMenuDropdown Checkmark DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-grid slds-grid--pull-padded slds-grid--vertical-align-center\\">
  <div class=\\"slds-col--padded\\"><span>Checkmark </span>
    <div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"Hk-szsQTVf\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled ignore-click-Hk-szsQTVf\\" tabindex=\\"0\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg></button></div>
  </div>
  <div class=\\"slds-col--padded\\"><span>Checkmark with right icon </span>
    <div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"Hkfifsm64G\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-more slds-button--icon-large ignore-click-Hkfifsm64G\\" tabindex=\\"0\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#settings\\"></use></svg><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg><span class=\\"slds-assistive-text\\">Checkmark with right icon</span></button></div>
  </div>
</div>"
`;

exports[`SLDSMenuDropdown CustomTrigger DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger--click"
  id="BymjMjm6VG"
  onBlur={undefined}
  onClick={[Function]}
  onFocus={null}
  onKeyDown={[Function]}
  onMouseDown={undefined}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={false}
    aria-haspopup={true}
    className="slds-button slds-button--neutral ignore-click-BymjMjm6VG"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex="0"
    title={undefined}
    type="button"
  >
    Dropdown
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--right"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
  </button>
</div>
`;

exports[`SLDSMenuDropdown CustomTrigger DOM & HTML Snapshots look the same 2`] = `"<div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"SyNsfo7pEG\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--neutral ignore-click-SyNsfo7pEG\\" tabindex=\\"0\\" type=\\"button\\">Dropdown<svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--right\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg></button></div>"`;

exports[`SLDSMenuDropdown Default DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger--click"
  id="ryHjfi764M"
  onBlur={undefined}
  onClick={[Function]}
  onFocus={null}
  onKeyDown={[Function]}
  onMouseDown={undefined}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={false}
    aria-haspopup={true}
    className="slds-button slds-button--icon-border-filled ignore-click-ryHjfi764M"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex="0"
    title={undefined}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
  </button>
</div>
`;

exports[`SLDSMenuDropdown Default DOM & HTML Snapshots look the same 2`] = `"<div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"SyIoMi7p4G\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled ignore-click-SyIoMi7p4G\\" tabindex=\\"0\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg></button></div>"`;

exports[`SLDSMenuDropdown DefaultIconLabel DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger--click"
  id="ryviGj7pNG"
  onBlur={undefined}
  onClick={[Function]}
  onFocus={null}
  onKeyDown={[Function]}
  onMouseDown={undefined}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={false}
    aria-haspopup={true}
    className="slds-button slds-button--neutral ignore-click-ryviGj7pNG"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex="0"
    title={undefined}
    type="button"
  >
    Dropdown
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--right"
    >
      <use
        xlinkHref={undefined}
      />
    </svg>
  </button>
</div>
`;

exports[`SLDSMenuDropdown DefaultIconLabel DOM & HTML Snapshots look the same 2`] = `"<div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"SJdjfjXT4f\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--neutral ignore-click-SJdjfjXT4f\\" tabindex=\\"0\\" type=\\"button\\">Dropdown<svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--right\\"><use></use></svg></button></div>"`;

exports[`SLDSMenuDropdown SubHeading DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger--click"
  id="SJFjfom6Ef"
  onBlur={undefined}
  onClick={[Function]}
  onFocus={null}
  onKeyDown={[Function]}
  onMouseDown={undefined}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={false}
    aria-haspopup={true}
    className="slds-button slds-button--icon-border-filled ignore-click-SJFjfom6Ef"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex="0"
    title={undefined}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
  </button>
</div>
`;

exports[`SLDSMenuDropdown SubHeading DOM & HTML Snapshots look the same 2`] = `"<div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"H1qoziXaEG\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled ignore-click-H1qoziXaEG\\" tabindex=\\"0\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg></button></div>"`;
